File: backend/app/Modules/Vehiculo/Models/VehiculoGasto.php
<?php

namespace App\Modules\Vehiculo\Models;

use App\Models\User;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class VehiculoGasto extends Model
{
    protected $table = 'vehiculo_gastos';

    protected $fillable = [
        'vehiculo_id',
        'tipo_gasto',
        'fecha_gasto',
        'concepto',
        'cantidad',
        'unidad_medida',
        'precio_unitario',
        'monto_total',
        'comprobantes',
        'registrado_por',
    ];

    protected $casts = [
        'fecha_gasto'     => 'date',
        'cantidad'        => 'decimal:2',
        'precio_unitario' => 'decimal:2',
        'monto_total'     => 'decimal:2',
        'comprobantes'    => 'array',
    ];

    // ─── Relaciones ───────────────────────────────────────────────

    public function vehiculo(): BelongsTo
    {
        return $this->belongsTo(Vehiculo::class, 'vehiculo_id');
    }

    public function registrador(): BelongsTo
    {
        return $this->belongsTo(User::class, 'registrado_por');
    }
}

File: frontend/lib/modules/flota/models/vehiculo_gasto.dart
class VehiculoGasto {
  final int id;
  final int vehiculoId;
  final String? tipoGasto;
  final DateTime? fechaGasto;
  final String concepto;
  final double cantidad;
  final String? unidadMedida;
  final double precioUnitario;
  final double montoTotal;
  final Map<String, dynamic>? vehiculo;

  VehiculoGasto({
    required this.id,
    required this.vehiculoId,
    this.tipoGasto,
    this.fechaGasto,
    required this.concepto,
    required this.cantidad,
    this.unidadMedida,
    required this.precioUnitario,
    required this.montoTotal,
    this.vehiculo,
  });

  factory VehiculoGasto.fromJson(Map<String, dynamic> json) {
    return VehiculoGasto(
      id: json['id'],
      vehiculoId: json['vehiculo_id'],
      tipoGasto: json['tipo_gasto'],
      fechaGasto: json['fecha_gasto'] != null
          ? DateTime.parse(json['fecha_gasto'])
          : null,
      concepto: json['concepto'] ?? '',
      cantidad: json['cantidad'] != null
          ? double.tryParse(json['cantidad'].toString()) ?? 0.0
          : 0.0,
      unidadMedida: json['unidad_medida'],
      precioUnitario: json['precio_unitario'] != null
          ? double.tryParse(json['precio_unitario'].toString()) ?? 0.0
          : 0.0,
      montoTotal: json['monto_total'] != null
          ? double.tryParse(json['monto_total'].toString()) ?? 0.0
          : 0.0,
      vehiculo: json['vehiculo'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'tipo_gasto': tipoGasto,
      'concepto': concepto,
      'cantidad': cantidad,
      'unidad_medida': unidadMedida,
      'precio_unitario': precioUnitario,
    };
  }
}

